contextualized/regression/lightning_modules.py

```
@@ -65,19 +85,134 @@ def __init__(
+        :param context_encoder_type: Type of encoder to use for the context (default is "mlp")
+        :param context_width: Width of the context encoder (default is 25)
+        :param context_layers: Number of layers in the context encoder (default is 1)
+        :param context_link_fn: Link function to apply to the output of the context encoder (default is "softmax")
+        :param task_encoder_type: Type of encoder to use for the task (default is "mlp")
+        :param task_width: Width of the task encoder (default is 25)
+        :param task_layers: Number of layers in the task encoder (default is 1)
+        :param task_link_fn: Link function to apply to the output of the task encoder (default is "identity")
 
         """
         # builds the metamodel
-        kwargs["univariate"] = False
-        self.metamodel = SINGLE_TASK_METAMODELS[self.metamodel_type](*args, **kwargs)
+        # no tasksplit metamodel here
+        meta_cls = SINGLE_TASK_METAMODELS[self.metamodel_type]
+        meta_params = inspect.signature(meta_cls).parameters
+        full_args = {
+            "context_dim": context_dim,
+            "x_dim": x_dim,
+            "y_dim": y_dim,
+            "univariate": univariate,
+            "encoder_type": encoder_type,
+            "width": width,
+            "layers": layers,
+            "link_fn": link_fn,
+            "num_archetypes": num_archetypes,
+            "context_archetypes": context_archetypes,
+            "task_archetypes": task_archetypes,
+            "context_encoder_type": context_encoder_type,
+            "context_width": context_width,
+            "context_layers": context_layers,
+            "context_link_fn": context_link_fn,
+            "task_encoder_type": task_encoder_type,
+            "task_width": task_width,
+            "task_layers": task_layers,
+            "task_link_fn": task_link_fn,
+        }
+        filtered_args = {k: v for k, v in full_args.items() if k in meta_params}
```

This also fails silently. Arg filtering is what we are trying to get rid of.